src/component/requests.js

```diff
@@ -15,32 +17,69 @@ export default class Requests extends React.Component {
     });
   }
 
+  componentDidMount() {
+    const requests = React.findDOMNode(this);
+    requests.addEventListener('scroll', () => {
+      const fromIndex = Math.ceil(requests.scrollTop / requestElementHeight) - 15;
```

Comment:
Have you checked how much time is spend on DOM manipulation in on frame while scrolling? I wonder if `scrollTop` triggers reflow. Think about using a Library that handles all that stuff. Listening on "scroll" usually is a performance killer.